controllers/UserController.js

```diff
@@ -0,0 +1,100 @@
+import UserModel from '../models/User.js'
+import jwt from 'jsonwebtoken'
+import bcrypt from 'bcrypt'
+import { validationResult } from 'express-validator';
+
+export const register = async (req, res) => {
+    try {
+        const errors = validationResult(req);
+        if(!errors.isEmpty()) {
+            return res.status(400).json(errors.array());
+        }
+
+        const password = req.body.password;
+        const salt = await bcrypt.genSalt(10);
+        const hash = await bcrypt.hash(password, salt);
+        const doc = new UserModel({
+            fullName: req.body.fullName,
+            email: req.body.email,
+            passwordHash: hash,
+        });
```

Comment:
Destructure the req.body object to extract fullName, email, and password.
```suggestion
        const { fullName, email, password } = req.body;
        
        const salt = await bcrypt.genSalt(10);
        const hash = await bcrypt.hash(password, salt);
        const doc = new UserModel({
            fullName,
            email,
            passwordHash: hash,
        });
```